openmls/src/test_utils/mod.rs

```diff
@@ -57,27 +67,144 @@ pub(crate) fn hex_to_bytes(hex: &str) -> Vec<u8> {
 
 /// Convert a hex string to a byte vector.
 /// If the input is `None`, this returns an empty vector.
-pub(crate) fn hex_to_bytes_option(hex: Option<String>) -> Vec<u8> {
+pub fn hex_to_bytes_option(hex: Option<String>) -> Vec<u8> {
     match hex {
         Some(s) => hex_to_bytes(&s),
         None => vec![],
     }
 }
 
+// === Print debug information about ciphersuites & backends ===
+
+#[allow(unused_macros)]
+macro_rules! debug_show_ciphersuite {
+    ($ciphersuite:ident) => {
+        println!("Using ciphersuite: {}", $ciphersuite.name());
+    };
+}
+
 #[allow(unused_macros)]
-macro_rules! ctest_ciphersuites {
-    ($name:ident, test($param_name:ident: $t:ty) $body:block) => {
-        test_macros::ctest!(
-            $name
-            [
-                CiphersuiteName::MLS10_128_DHKEMX25519_AES128GCM_SHA256_Ed25519,
-                CiphersuiteName::MLS10_128_DHKEMP256_AES128GCM_SHA256_P256,
-                CiphersuiteName::MLS10_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519,
-            ]
-            {
-                fn test($param_name: $t) $body
-                test(param)
-            }
+macro_rules! debug_show_backend {
+    ($backend:ident) => {
+        #[inline]
+        fn type_of<T>(_: &T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        println!("Using backend: {}", type_of($backend));
+    };
+}
+
+#[allow(unused_macros)]
+macro_rules! debug_show_backend_and_ciphersuite {
+    ($backend:ident, $ciphersuite:ident) => {
+        #[inline]
+        fn type_of<T>(_: &T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        println!(
+            "Using backend: {}, ciphersuite: {}",
+            type_of($backend),
+            $ciphersuite.name()
         );
     };
 }
+
+// === Define backend per platform ===
+
+// For now we only use Evercrypt on specific platforms and only if the feature was enabled
+
+#[cfg(all(
+    target_arch = "x86_64",
+    not(target_os = "macos"),
+    not(target_family = "wasm"),
+    feature = "evercrypt",
+))]
+pub use evercrypt_backend::OpenMlsEvercrypt;
+
+// This backend is currently used on all platforms
+pub use openmls_rust_crypto::OpenMlsRustCrypto;
+
+// === Backends ===
+
+#[cfg(any(
+    not(target_arch = "x86_64"),
+    target_os = "macos",
+    target_family = "wasm",
+    not(feature = "evercrypt")
+))]
+#[template]
+#[rstest(backend,
+    case(&OpenMlsRustCrypto::default()),
```

Comment:
Can you name all the cases please? :) Otherwise it's impossible to read or filter the tests.
Something like the suggestion here for example (and similar for all the other cases).
```suggestion
    case::rust_crypto(&OpenMlsRustCrypto::default()),
```